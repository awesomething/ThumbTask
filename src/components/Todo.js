import React from 'react'
import config from '../config'

const Todo = ({text, todo, todos, setTodos}) => {
    //events
    const deleteTodo = () => {
       return fetch(`${config.API_ENDPOINT}/todos`, {
          method: "DELETE",
          headers: {
            "content-type": "application/json"
          },
        })
          .then((res) => {
            if (!res.ok) {
              return res.json().then((error) => Promise.reject(error));
            }
            return res;
          })
          .catch((error) => {
            console.log(error);
          });
      };

    const deleteHandler =() => {
        deleteTodo(todo)
      setTodos(todos.filter(el => el.id !== todo.id));
    };
    const completeHandler = () => {
        setTodos(todos.map(item => {
            if(item.id === todo.id){
                return {
                    ...item, completed: !item.completed
                }
            }
            return item;
        }))
    }
    return (
        <div className="todo">
            <li className={`todo-item ${todo && todo.completed ? "completed" : " "}`}>{text}</li>
            <button onClick={completeHandler} className="complete-btn"><i className="fas fa-check"></i></button>
            <button onClick={deleteHandler} className="trash-btn"><i className="fas fa-trash"></i></button>
        </div>
    );
}

export default Todo;