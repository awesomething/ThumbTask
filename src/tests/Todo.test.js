import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Todo from '../components/Todo'
import {configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })


describe(`Todo component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    
  }

  it('renders a .Todo by default', () => {
    const wrapper = shallow(<Todo />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Todo given props', () => {
    const wrapper = shallow(<Todo {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
