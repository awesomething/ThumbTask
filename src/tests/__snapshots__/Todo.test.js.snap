// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders a .Todo by default 1`] = `
<div
  className="todo"
>
  <li
    className="todo-item  "
  />
  <button
    className="complete-btn"
    onClick={[Function]}
  >
    <i
      className="fas fa-check"
    />
  </button>
  <button
    className="trash-btn"
    onClick={[Function]}
  >
    <i
      className="fas fa-trash"
    />
  </button>
</div>
`;

exports[`Todo component renders the Todo given props 1`] = `
<div
  className="todo"
>
  <li
    className="todo-item  "
  />
  <button
    className="complete-btn"
    onClick={[Function]}
  >
    <i
      className="fas fa-check"
    />
  </button>
  <button
    className="trash-btn"
    onClick={[Function]}
  >
    <i
      className="fas fa-trash"
    />
  </button>
</div>
`;
